src/yubikey.rs

```diff
@@ -648,7 +647,11 @@ impl YubiKey {
             return Err(Error::AuthenticationError);
         }
 
-        Ok(response.data()[4..12].try_into().unwrap())
```

Comment:
Nothing actually checked `response.data().len()` here so this could've potentially panicked if the response were malformed